### PR TITLE
Fix: overall improvements

### PR DESCRIPTION
**Pull request by maslick**
* Add mocka tests
* ~~Make examples work~~
* Small optimisations

### CODE REVIEW

File: examples/encode.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
-const encode = require('upnqr').encode
+const {encode} = require('../lib/upnqr');
```
Comment (matjaz) [inline examples/encode.js]: leave `require('upnqr')` so it can be copy/pasted to your project.

File: examples/encode.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
 
 const upn = {
+  slog: 'UPNQR',
```
Comment (matjaz) [inline examples/encode.js]: this is default/only value. This should be fixed in lib.

File: examples/encode.js [hunk 3]
```diff
@@ -17,8 +18,7 @@ const upn = {
   ulica_prejemnika: 'Povšetova ulica 6',
   kraj_prejemnika: '1000 Ljubljana',
   rezerva: 'dodatek do skupaj 411 znakov'
-}
+};
```
Comment (matjaz) [inline examples/encode.js]: take  a look at https://standardjs.com

File: package.json [hunk 4]
```diff
@@ -13,7 +16,8 @@
     "lib/*"
   ],
   "scripts": {
-    "test": "standard"
+    "test": "mocha test",
```
Comment (matjaz) [inline package.json:19]: before testing also check code style `"test": "standard && mocha test"`

File: package.json [hunk 5]
```diff
@@ -13,7 +16,8 @@
...
+    "standard": "standard"
```
Comment (matjaz) [inline package.json]: remove this line. use `npx standard`

**Review by matjaz — CHANGES_REQUESTED** [review]
Thanks for contributing. We use https://standardjs.com
I suggest to make multiple PR, for fixes, tests, optimisations.

File: test.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+const {encode, decode} = require('./lib/upnqr');
```
Comment (matjaz) [inline test.js]: move this file to test folder

File: package.json (hunk 5 above)
Comment (maslick) [reply package.json]: I would separate testing from linting... But it's up to you ;)

File: lib/util.js [hunk 7]
```diff
@@ -1,73 +1,53 @@
...
-  if (parts.length !== 3) {
-    error('Invalid date format')
-  }
+  if (parts.length !== 3) error('Invalid date format')
   const d = new Date()
   d.setTime(0)
   d.setDate(num(parts[0]))
   d.setMonth(num(parts[1]) - 1)
-  d.setYear(num(parts[2]))
+  d.setFullYear(num(parts[2]))
```
Comment (matjaz) [inline lib/util.js:33]: nice catch

**Comment by matjaz** [comment]
Thanks for contributing!

**Pull request merged**